### PR TITLE
Add throwing inaccuracy

### PR DESCRIPTION
**Pull request by notquitehadouken**
## About the PR

Made throwing have a wee bit of inaccuracy

## Why / Balance

Because throwing a lot of things at once is usually useful, but doing it with perfect accuracy on each throw doesn't make a lot of sense.

## Technical details

Rotates the throw vector depending on how long it has been since the last throw

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/45953835/8462c56f-fc39-400d-ab1b-5ab79b85b20f

## Breaking changes

**Changelog**

:cl:
- add: Throwing gets rapidly decaying inaccuracy for about one second after you throw.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
what's the purpose in reducing the cooldown from 0.5 -> 0.125? It was implemented specifically to mitigate throwing abuse. Yes, there's an accuracy penalty, but you could still spam throw objects in close combat to reliably hit way quicker than you could otherwise.

**Comment by notquitehadouken** [comment]
> what's the purpose in reducing the cooldown from 0.5 -> 0.125? It was implemented specifically to mitigate throwing abuse. Yes, there's an accuracy penalty, but you could still spam throw objects in close combat to reliably hit way quicker than you could otherwise.

cooldown was reduced so there wasn't just No Penalty (since it gets perfect accuracy at 0.5s)
i can raise it if need be

**Comment by GreyMario** [comment]
Raise the cooldown to 0.5 seconds and remove the accuracy penalty.

**Comment by notquitehadouken** [comment]
> Raise the cooldown to 0.5 seconds and remove the accuracy penalty.

this would literally just be closing the pr

**Comment by Arendian** [comment]
For the janitor's sanity, if this gets added the random chance to miss the disposal bin should probably be removed.

**Comment by notquitehadouken** [comment]
> For the janitor's sanity, if this gets added the random chance to miss the disposal bin should probably be removed.

doesn't apply to single throws :trollface:

**Comment by keronshb** [comment]
While I don't agree with the initial throw cooldown, it's there as a bandaid to prevent spam throwing exactly like this.

I still believe having a throw mode and having a stamina cost per throw (with a more clear stamina meter) is a better solution. Since throw mode also allows you to catch things.

Inaccuracy doesn't work too well when the gap is shorter, which throwing an obscene amount of batons and other items with one button combination was one of the main issues.,

**Comment by notquitehadouken** [comment]
give me a second while i add media again

File: Content.Server/Hands/Systems/HandsSystem.cs [hunk 1]
```diff
@@ -204,6 +205,16 @@ public bool ThrowHeldItem(EntityUid player, EntityCoordinates coordinates, float
...
 
+            if (_timing.CurTime < hands.LastThrow + hands.TimeToPerfectThrow)
+            {
+                var timeSinceLast = _timing.CurTime - hands.LastThrow;
+                var inaccuracyModifier = 1 - (timeSinceLast / hands.TimeToPerfectThrow);
+                inaccuracyModifier *= inaccuracyModifier;
+                var maxInaccuracy = hands.InaccuracyAfterThrow * inaccuracyModifier;
+                var totalInaccuracy = _random.NextAngle(-maxInaccuracy, maxInaccuracy);
+                direction = totalInaccuracy.RotateVec(direction);
+            }
```
Comment (metalgearsloth) [inline Content.Server/Hands/Systems/HandsSystem.cs:216]: You're doubling the throwing angle so 180 degrees actually means you can throw things behind you as it's 180 degrees either direction.

File: Content.Shared/Hands/Components/HandsComponent.cs [hunk 2]
```diff
@@ -59,16 +59,28 @@ public sealed partial class HandsComponent : Component
...
+    ///     How long it takes for accuracy to become perfect.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan TimeToPerfectThrow = TimeSpan.FromSeconds(1);
+
+    /// <summary>
+    ///     The absolute minimum time inbetween throws.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan ThrowCooldown = TimeSpan.FromSeconds(0.25f);
```
Comment (metalgearsloth) [inline Content.Shared/Hands/Components/HandsComponent.cs:83]: 0.5 was fine, 0.25 is too low.

Comment (notquitehadouken) [reply Content.Shared/Hands/Components/HandsComponent.cs]: done

**Pull request closed without merging**